linkerd/tls/src/main/scala/io/l5d/clientTls/boundPath.scala

```diff
@@ -32,6 +32,7 @@ case class boundPath(caCertPath: Option[String], names: Seq[NameMatcherConfig],
       } match {
         case None if (strict.getOrElse(true)) =>
           throw new MatcherError(s"Unable to match ${path.show} with available names: ${(names.map(_.prefix)).mkString(",")}")
+        case default => default
```

Comment:
I tried that in my previous review, but the yield block never executes if commonName == None